src/Tools/BuildValidator/CompilationDiff.cs

```diff
@@ -87,58 +144,118 @@ public static CompilationDiff CreatePlaceholder(FileInfo originalBinaryPath, boo
                     logger.LogError(diag.ToString());
                 }
 
-                return new CompilationDiff(emitResult.Diagnostics, originalBinaryPath.FullName);
+                return new CompilationDiff(
+                    assemblyInfo,
+                    CompilationOutcome.CompilationError,
+                    diagnostics: emitResult.Diagnostics);
             }
             else
             {
-                var originalBytes = File.ReadAllBytes(originalBinaryPath.FullName);
+                var originalBytes = File.ReadAllBytes(assemblyInfo.FilePath);
```

Comment:
should we add a note that we want to avoid doing I/O here? e.g. eventually we need to make the public caller pass in a PE stream or pass in the assembly fully read into memory.

Reply:
no need to change the PR for this--but this is a good moment to decide if we need to file an issue

Reply:
Suddenly it occurs to me that we're in the CLI tool's area now, so it's not as much of a concern--but if we end up pulling parts of this over into the library side to help users diagnose rebuild failures, I think we'll have to keep an eye out for this sort of thing.

Reply:
Agree we need to keep an eye out for this. Really need to put a hard stop on anything like this going into the library.